docs/api-reference/types/command.md

```diff
@@ -0,0 +1,163 @@
+# Command
+
+[← Back to API Reference](../README.md) | [← Back to Table of Contents](../../README.md)
+
+`Command` is a base class in Reactive Domain that implements the `ICorrelatedMessage` interface and serves as the foundation for all command messages in the system.
+
+## Overview
+
+Commands in Reactive Domain represent requests for the system to perform an action. They are part of the write side of the CQRS pattern and typically result in state changes. The `Command` base class provides common functionality for all command implementations, including correlation and causation tracking.
+
+## Class Definition
+
+```csharp
+public abstract class Command : ICommand, ICorrelatedMessage
+{
+    public Guid MsgId { get; }
+    public Guid CorrelationId { get; }
+    public Guid CausationId { get; }
+    
+    protected Command()
+    {
+        MsgId = Guid.NewGuid();
+        CorrelationId = MsgId;
+        CausationId = MsgId;
+    }
+    
+    protected Command(Guid correlationId, Guid causationId)
+    {
+        MsgId = Guid.NewGuid();
+        CorrelationId = correlationId;
+        CausationId = causationId;
+    }
+}
+```
+
+## Key Features
+
+- **Message Identity**: Provides a unique `MsgId` for each command
+- **Correlation Tracking**: Implements `ICorrelatedMessage` for tracking related messages
+- **Immutability**: Ensures commands are immutable after creation
+- **Type Safety**: Provides a type-safe base for all command implementations
+
+## Usage
+
+### Defining a Command
+
+To create a new command type, inherit from the `Command` base class:
+
+```csharp
+public class CreateAccount : Command
+{
+    public readonly Guid AccountId;
+    public readonly string AccountNumber;
+    public readonly string CustomerName;
+    
+    public CreateAccount(Guid accountId, string accountNumber, string customerName)
+        : base()
+    {
+        AccountId = accountId;
+        AccountNumber = accountNumber;
+        CustomerName = customerName;
+    }
+    
+    // Constructor for correlated commands
+    public CreateAccount(Guid accountId, string accountNumber, string customerName, 
+                         Guid correlationId, Guid causationId)
+        : base(correlationId, causationId)
+    {
```

Comment:
This overload of the base ctor should not be used explicitly. Use the `MessageBuilder` factory to set correlation and causation IDs. Most classes that inherit from `Command` will implicitly use the default ctor.